src/main/java/Ui.java

```diff
@@ -0,0 +1,125 @@
+import java.util.ArrayList;
+
+/**
+ * Deals with interactions with the user.
+ */
+public class Ui {
+    public static String TAB = "     ";
+    public static String LINE = "     ............................................................";
+
+    public Ui() { }
+
+    /**
+     * Prints Elly logo and introduction message.
+     */
+    public void printIntro() {
+        String logo = " _____   _   _\n"
+                + "| ____| | | | |\n"
+                + "| |___  | | | | __   __\n"
+                + "|  ___| | | | | \\ \\ / /\n"
+                + "| |___  | | | |  \\ v /\n"
+                + "|_____| |_| |_|  /  /\n"
+                + "                /__/\n";
+
+        System.out.println("   C H A T   W I T H\n" + logo);
+
+        System.out.println(LINE + "\n"
+                + TAB + "Hi there! I'm Elly.\n"
+                + TAB + "How can I help you today?\n"
+                + LINE);
+    }
+
+    /**
+     * Prints line for Elly's response display.
+     */
+    public void printLine() {
+        System.out.println(LINE);
+    }
+
+    public String printList(ArrayList<Task> tasks, int numTasks) {
+        StringBuilder response;
+        if (numTasks == 0) {
+            response = new StringBuilder("Yay! There are no tasks in your list.");
+        } else if (numTasks < 0) {  // print filtered tasks
+            response = new StringBuilder("Here are the matching tasks in your list:\n");
```

Comment:
Hi , maybe you should create a named constant for these responses for better readability as recommended in Section 5.4e